src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -22,6 +22,23 @@
 from pants.rules.core.strip_source_root import SourceRootStrippedSources
 
 
+def get_timeout_seconds_for_target(test_target_timeout, timeout_default, timeout_maximum):
+  """Calculate the timeout for a test target.
+
+  If a target has no timeout configured its timeout will be set to the default timeout.
+
+  :param test_target: A test target
+  :param timeout_default: the default timeout in seconds.
+  :param timeout_maximum: the maximum timeout in seconds.
+  :return: timeout in seconds.
+  """
+  if test_target_timeout is None:
+    test_target_timeout = timeout_default
+  if timeout_maximum is not None and test_target_timeout > timeout_maximum:
+    return timeout_maximum
+  return test_target_timeout
```

Comment:
Can simplify:

```
return max(test_target_timeout, timeout_maximum or -1)
```

If `timeout_maximum` is None, then `test_target_timeout` will always win because it's presumably a positive number, so > -1.

Reply:
Test_target_timeout may be None, so this would have to be `max(test_target_timeout or -1, timeout_maximum or -1)` I then still have to check if timeout_maximum is greater than the result (so I know if I should override it. 

I think this is an improvement though:

```
  if timeout_maximum_seconds is not None and test_target_timeout_seconds is not None:
    return min(timeout_maximum_seconds, test_target_timeout_seconds)
```